test/files/jvm/serialization-new.check

```diff
@@ -112,8 +112,8 @@ x = TreeSet(1, 2, 3)
 y = TreeSet(1, 2, 3)
 x equals y: true, y equals x: true
 
-x = LazyList(_, ?)
-y = LazyList(_, ?)
+x = LazyList(?)
+y = LazyList(?)
```

Comment:
The implementation of `addString` is very hard to review. I guess all the places where you have changed the expected output of the tests are places where we previously knew that the lazy list was not empty and now we don’t know?

Reply:
I agree that it's hard to review (it was also very hard to understand well enough to change to not use `tailDefined`). The tests were changed either because (a) previously we knew it was not empty, but now we don't, or (b) previously the tail hadn't been evaluated to a cycle, but since the tail is no longer lazy (since the whole thing is lazy), it now knows it's cyclic slightly earlier